engine/execution/storehouse.go

```diff
@@ -0,0 +1,74 @@
+package execution
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// RegisterStore is the interface for register store
+// see implementation in engine/execution/storehouse/register_store.go
+type RegisterStore interface {
+	// GetRegister first try to get the register from InMemoryRegisterStore, then OnDiskRegisterStore
+	GetRegister(height uint64, blockID flow.Identifier, register flow.RegisterID) (flow.RegisterValue, error)
+
+	// SaveRegisters saves to InMemoryRegisterStore first, then trigger the same check as OnBlockFinalized
+	// Depend on InMemoryRegisterStore.SaveRegisters
+	SaveRegisters(header *flow.Header, registers []flow.RegisterEntry) error
+
+	// Depend on FinalizedReader's GetFinalizedBlockIDAtHeight
```

Comment:
what is the purpose of this "Depend" comments? I'm not sure I understand how is this needed

Reply:
I used Depends to list it's dependencies. 

For instance, the OnBlockFinalized method depends on FinalizedReader, ExecutedFinalizedWAL, and OnDiskRegisterStore to work. but why OnBlockFinalized depends on them? Because it will first call FinalizedReader's GetFinalizedBlockIDAtHeight to find the next unfinalized and executed block, and then if it's executed, then it will `Append` to `ExecutedFinalizedWAL`, and then `SaveRegisters` to the `OnDiskRegisterStore`. 

The interfaces of `FinalizedReader`, `ExecutedFinalizedWAL` and `OnDiskRegisterStore` are all defined in this interface file, so it's easy to find the behavior of those methods, which could help understand the behavior of `OnBlockFinalized` method